### PR TITLE
operators community-windows-machine-config-operator (5.1.0)

### PR DESCRIPTION
**Pull request by alinaryan**
Signed-off-by: Alina Ryan aliryan@redhat.com

Thanks submitting your Operator. Please check below list before you create your Pull Request.

### New Submissions

* [ ] Are you familiar with our [contribution guidelines](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-via-pr.md)?
* [ ] Have you [packaged and deployed](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) your Operator for Operator Framework?
* [ ] Have you tested your Operator with all Custom Resource Definitions?
* [ ] Have you tested your Operator in all supported [installation modes](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#operator-metadata)?
* [ ] Have you considered whether you want use [semantic versioning order](https://github.com/operator-framework/community-operators/blob/master/docs/operator-ci-yaml.md#semver-mode)?
* [ ] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-prerequisites.md#sign-your-work)?
* [ ] Is operator [icon](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#operator-icon) set?

### Updates to existing Operators

* [ ] Did you create a `ci.yaml` file according to the [update instructions](https://github.com/operator-framework/community-operators/blob/master/docs/operator-ci-yaml.md)?
* [ ] Is your new CSV pointing to the previous version with the `replaces` property if you chose `replaces-mode` via the `updateGraph` property in `ci.yaml`?
* [ ] Is your new CSV referenced in the [appropriate channel](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#channels) defined in the `package.yaml` or `annotations.yaml` ?
* [ ] Have you tested an update to your Operator when deployed via OLM?
* [ ] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-prerequisites.md#sign-your-work)?

### Your submission should not

* [ ] Modify more than one operator
* [ ] Modify an Operator you don't own
* [ ] Rename an operator - please remove and add with a different name instead
* [ ] Modify any files outside the above mentioned folders
* [ ] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [ ] Description about the managed Application and where to find more information
2. [ ] Features and capabilities of your Operator and how to use it
3. [ ] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [ ] Human readable name and 1-liner description about your Operator
* [ ] Valid [category name](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#categories)<sup>1</sup>
* [ ] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [ ] Links to the maintainer, source code and documentation
* [ ] Example templates for all Custom Resource Definitions intended to be used
* [ ] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

--

<sup>1</sup> If you feel your Operator does not fit any of the pre-defined categories, file an issue against this repo and explain your need

<sup>2</sup> For more information see [here](https://sdk.operatorframework.io/docs/overview/#operator-capability-level)

### CODE REVIEW

File: operators/community-windows-machine-config-operator/5.1.0/manifests/manager-role_rbac.authorization.k8s.io_v1_role.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (jrvaldes) [inline operators/community-windows-machine-config-operator/5.1.0/manifests/manager-role_rbac.authorization.k8s.io_v1_role.yaml:1]: Remove the placeholder from the commit message.

```
Signed-off-by: GITHUB_NAME <GITHUB_EMAIL>
```

File: operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    olm.skipRange: '>=4.0.0 <5.0.0'
+    operatorframework.io/cluster-monitoring: "true"
+    operatorframework.io/suggested-namespace: openshift-windows-machine-config-operator
+    operators.operatorframework.io/builder: operator-sdk-v1.15.0+git
+    operators.openshift.io/valid-subscription: '["Red Hat OpenShift support for Windows Containers"]'
+    operators.operatorframework.io/project_layout: go.kubebuilder.io/v3
+    repository: https://github.com/openshift/windows-machine-config-operator
+    support: Red Hat
+  name: community-windows-machine-config-operator-5.1.0
+  namespace: placeholder
```
Comment (jrvaldes) [inline operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml:20]: s/placeholder/openshift-windows-machine-config-operator

File: operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+                        valueFrom:
+                          fieldRef:
+                            fieldPath: metadata.annotations['olm.targetNamespaces']
+                      - name: POD_NAME
+                        valueFrom:
+                          fieldRef:
+                            fieldPath: metadata.name
+                      - name: OPERATOR_NAME
+                        value: community-windows-machine-config-operator
+                    image: ""
```
Comment (jrvaldes) [inline operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml:254]: Missing image URL

File: operators/community-windows-machine-config-operator/5.1.0/metadata/annotations.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+annotations:
+  # Core bundle annotations.
+  operators.operatorframework.io.bundle.mediatype.v1: registry+v1
+  operators.operatorframework.io.bundle.manifests.v1: manifests/
+  operators.operatorframework.io.bundle.metadata.v1: metadata/
+  operators.operatorframework.io.bundle.package.v1: windows-machine-config-operator
+  operators.operatorframework.io.bundle.channels.v1: preview,stable
```
[the remark this answers is not in the document]
Comment (alinaryan) [reply operators/community-windows-machine-config-operator/5.1.0/metadata/annotations.yaml:7]: Just preview. Fixed.

File: operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+    description: An operator that enables Windows container workloads on OCP
+    olm.skipRange: '>=4.0.0 <5.0.0'
+    operatorframework.io/cluster-monitoring: "true"
+    operatorframework.io/suggested-namespace: openshift-windows-machine-config-operator
+    operators.operatorframework.io/builder: operator-sdk-v1.15.0+git
+    operators.openshift.io/valid-subscription: '["Red Hat OpenShift support for Windows Containers"]'
+    operators.operatorframework.io/project_layout: go.kubebuilder.io/v3
+    repository: https://github.com/openshift/windows-machine-config-operator
+    support: Red Hat
+  name: community-windows-machine-config-operator-5.1.0
```
Comment (mtnbikenc) [inline operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml:19]: Based on past CSVs, the format is slightly different:
```suggestion
  name: community-windows-machine-config-operator.v5.1.0
```

File: operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+    capabilities: Seamless Upgrades
+    categories: OpenShift Optional
+    certified: "false"
+    createdAt: "2022-06-01T09:47:21Z"
+    description: An operator that enables Windows container workloads on OCP
+    olm.skipRange: '>=4.0.0 <5.0.0'
+    operatorframework.io/cluster-monitoring: "true"
+    operatorframework.io/suggested-namespace: openshift-windows-machine-config-operator
+    operators.operatorframework.io/builder: operator-sdk-v1.15.0+git
+    operators.openshift.io/valid-subscription: '["Red Hat OpenShift support for Windows Containers"]'
```
Comment (mtnbikenc) [inline operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml:15]: Should this line should be removed from the community CSV?

Comment (alinaryan) [reply operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml:15]: yes, will remove

File: operators/community-windows-machine-config-operator/5.1.0/metadata/annotations.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  operators.operatorframework.io.bundle.mediatype.v1: registry+v1
+  operators.operatorframework.io.bundle.manifests.v1: manifests/
+  operators.operatorframework.io.bundle.metadata.v1: metadata/
+  operators.operatorframework.io.bundle.package.v1: windows-machine-config-operator
+  operators.operatorframework.io.bundle.channels.v1: preview
+  operators.operatorframework.io.bundle.channel.default.v1: preview
+  operators.operatorframework.io.metrics.mediatype.v1: metrics+v1
+  operators.operatorframework.io.metrics.builder: operator-sdk-v2.0.0+git
+  operators.operatorframework.io.metrics.project_layout: go.kubebuilder.io/v3
+
```
[the remark this answers is not in the document]
Comment (alinaryan) [reply operators/community-windows-machine-config-operator/5.1.0/metadata/annotations.yaml:13]: Added

File: operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml (hunk 2 above)
Comment (alinaryan) [reply operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml:19]: we will actually need 'placeholder'. Doc ref: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#csv-metadata

File: operators/community-windows-machine-config-operator/5.1.0/metadata/annotations.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  operators.operatorframework.io.bundle.mediatype.v1: registry+v1
+  operators.operatorframework.io.bundle.manifests.v1: manifests/
+  operators.operatorframework.io.bundle.metadata.v1: metadata/
+  operators.operatorframework.io.bundle.package.v1: windows-machine-config-operator
+  operators.operatorframework.io.bundle.channels.v1: preview
+  operators.operatorframework.io.bundle.channel.default.v1: preview
+  operators.operatorframework.io.metrics.mediatype.v1: metrics+v1
+  operators.operatorframework.io.metrics.builder: operator-sdk-v2.0.0+git
+  operators.operatorframework.io.metrics.project_layout: go.kubebuilder.io/v3
+  com.redhat.openshift.versions: =v4.10
```
Comment (mtnbikenc) [inline operators/community-windows-machine-config-operator/5.1.0/metadata/annotations.yaml:12]:
```suggestion
  com.redhat.openshift.versions: "v4.10"
```

Comment (sebsoto) [reply operators/community-windows-machine-config-operator/5.1.0/metadata/annotations.yaml:12]: i believe `=4.10` is correct
thats what was put for 5.0.0

Comment (mtnbikenc) [reply operators/community-windows-machine-config-operator/5.1.0/metadata/annotations.yaml:12]: Ah, ok.  I was just going by what was requested [here](https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/1262#discussion_r887655229).

File: operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml (hunk 3 above)
Comment (alinaryan) [reply operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml]: Added

File: operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml (hunk 5 above)
Comment (alinaryan) [reply operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml:18]: Added

File: operators/community-windows-machine-config-operator/5.1.0/metadata/annotations.yaml (hunk 8 above)
Comment (mtnbikenc) [reply operators/community-windows-machine-config-operator/5.1.0/metadata/annotations.yaml:12]: Yep, `=4.10` is correct per [link](https://redhat-connect.gitbook.io/certified-operator-guide/ocp-deployment/operator-metadata/bundle-directory/managing-openshift-versions).

**Review by sebsoto — APPROVED** [review]

File: operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml (hunk 5 above)
Comment (mtnbikenc) [reply operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml:18]: Is this coming in a commit you haven't pushed yet?

Comment (alinaryan) [reply operators/community-windows-machine-config-operator/5.1.0/manifests/community-windows-machine-config-operator.5.1.0.clusterserviceversion.yaml]: Thanks for catching that - I think it reverted back by accident before one of my last pushes

**Comment by mtnbikenc** [comment]
I guess @sebsoto 's [approval](https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/1262#pullrequestreview-995441508) didn't actually set the status to Approved.  Maybe it is time to [join the org](https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/1262#issuecomment-1143650143) to get this label added automatically?

**Comment by selansen** [comment]
if we can have write access, then we can merge it as well. not sure who can add us!

**Comment by mvalarh** [comment]
@alinaryan @selansen @mtnbikenc You can add yourself to `reviewer` list in `ci.yaml` and next PR can be automerged. More info : https://redhat-openshift-ecosystem.github.io/community-operators-prod/operator-ci-yaml/#reviewers

**Pull request merged**